### PR TITLE
#87 add instance for time of day

### LINKED ISSUE #87: Add instances for TimeOfDay
See [`Data.Time.LocalTime.TimeOfDay`](https://hackage.haskell.org/package/time-1.6.0.1/docs/Data-Time-LocalTime.html#t:TimeOfDay).

### PR DESCRIPTION
**Pull request by anneloreegger**
fix #87

### CODE REVIEW

File: src/Data/Swagger/Internal/ParamSchema.hs [hunk 1]
```diff
@@ -193,6 +193,9 @@ timeParamSchema fmt = mempty
 instance ToParamSchema Day where
   toParamSchema _ = timeParamSchema "date"
 
+instance ToParamSchema TimeOfDay where
```
Comment (fizruk) [inline src/Data/Swagger/Internal/ParamSchema.hs]: Could you please add a doctest here?

File: test/Data/Swagger/ParamSchemaSpec.hs [hunk 2]
```diff
@@ -30,6 +34,7 @@ spec = do
     context "Unary records" $ do
       context "Email (unary record)"  $ checkToParamSchema (Proxy :: Proxy Email)  emailSchemaJSON
       context "UserId (non-record newtype)" $ checkToParamSchema (Proxy :: Proxy UserId) userIdSchemaJSON
+    context "TimeOfDay" $ checkToParamSchema (Proxy :: Proxy TimeOfDay) (Object (HM.fromList [("format",String "hh:MM:ss"),("type",String "string")]))
```
Comment (fizruk) [inline test/Data/Swagger/ParamSchemaSpec.hs]: Could you put schema's JSON value in `SpecCommon`, just like `colorSchemaJSON` and others?

File: test/Data/Swagger/SchemaSpec.hs [hunk 3]
```diff
@@ -110,6 +115,7 @@ spec = do
     context "MyRoseTree' (inlineNonRecursiveSchemas)" $ checkInlinedRecSchema (Proxy :: Proxy MyRoseTree') myRoseTreeSchemaJSON'
   describe "Bounded Enum key mapping" $ do
     context "ButtonImages" $ checkToSchema (Proxy :: Proxy ButtonImages) buttonImagesSchemaJSON
+    context "TimeOfDay" $ checkToSchema (Proxy :: Proxy TimeOfDay ) (Object (HM.fromList [("example",String "12:33:15"),("format",String "hh:MM:ss"),("type",String "string")]))
```
Comment (fizruk) [inline test/Data/Swagger/SchemaSpec.hs]: Could you put schema's JSON value in `SpecCommon`, just like `colorSchemaJSON` and others?

File: src/Data/Swagger/Internal/ParamSchema.hs (hunk 1 above)
Comment (anneloreegger) [reply src/Data/Swagger/Internal/ParamSchema.hs]: done :)

File: test/Data/Swagger/ParamSchemaSpec.hs (hunk 2 above)
Comment (anneloreegger) [reply test/Data/Swagger/ParamSchemaSpec.hs]: done

File: test/Data/Swagger/SchemaSpec.hs (hunk 3 above)
Comment (anneloreegger) [reply test/Data/Swagger/SchemaSpec.hs]: done

**Comment by anneloreegger** [comment]
Are there any news on this issue?

**Comment by fisx** [comment]
Could you please fix the merge conflict?  I have commit authority here now and will make sure this gets merged once you have.

Thanks!

**Comment by anneloreegger** [comment]
Sorry I'm very busy lately. But now I fixed the merge conflict. So it should hopefully be ready to be merged :)

File: test/Data/Swagger/SchemaSpec.hs [hunk 4]
```diff
@@ -105,6 +108,7 @@ spec = do
     context "MyRoseTree' (inlineNonRecursiveSchemas)" $ checkInlinedRecSchema (Proxy :: Proxy MyRoseTree') myRoseTreeSchemaJSON'
   describe "Bounded Enum key mapping" $ do
     context "ButtonImages" $ checkToSchema (Proxy :: Proxy ButtonImages) buttonImagesSchemaJSON
+    context "TimeOfDay" $ checkToSchema (Proxy :: Proxy Data.Time.LocalTime.TimeOfDay ) timeOfDaySchemaJSON
```
Comment (fisx) [inline test/Data/Swagger/SchemaSpec.hs]:
```suggestion
    context "TimeOfDay" $ checkToSchema (Proxy :: Proxy Data.Time.LocalTime.TimeOfDay) timeOfDaySchemaJSON
```

File: src/Data/Swagger/Internal/ParamSchema.hs [hunk 5]
```diff
@@ -193,6 +193,12 @@ timeParamSchema fmt = mempty
 instance ToParamSchema Day where
   toParamSchema _ = timeParamSchema "date"
 
+-- |
+-- >>> toParamSchema (Proxy :: Proxy TimeOfDay) ^. format
+-- Just "hh:MM:ss"
+instance ToParamSchema TimeOfDay where
+  toParamSchema _ = timeParamSchema "hh:MM:ss"
```
Comment (fisx) [inline src/Data/Swagger/Internal/ParamSchema.hs]: Why are `hh`, `ss` small, but `MM` is capitalized?  Could you explain this in the haddocks?

Comment (anneloreegger) [reply src/Data/Swagger/Internal/ParamSchema.hs]: Actually I am not sure if it makes a difference if MM is capitalized or not. I just took the schema that was used for LocalTime ("yyyy-mm-ddThh:MM:ss") and removed the parts I didn't need.
So basically I left the M capitalized for being consistent, but I don't know if there is another reason.

Comment (fizruk) [reply src/Data/Swagger/Internal/ParamSchema.hs]: `mm` is month. `MM` is minutes.

**Pull request merged**